File: esercizi_java/test/compito_java_2/Gestione.java
import java.util.List;
import java.util.LinkedList;
import java.util.HashMap;
import java.util.Map;
import java.util.StringTokenizer;
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class Gestione {
    static List<Dipendente> dipendente = new LinkedList<Dipendente>();
    static List<Cliente> cliente = new LinkedList<Cliente>();
    static Map<Integer, Dipendente> codProg = new HashMap<Integer, Dipendente>();

    public static void main(String[] args) {

        // punto 1
        try {
            BufferedReader br = new BufferedReader(new FileReader("dipendenti.txt"));
            String line = br.readLine();
            while (line != null) {
                StringTokenizer tok = new StringTokenizer(line);
                int cod = Integer.parseInt(tok.nextToken());
                String tipo = tok.nextToken();
                line = br.readLine();
                String nome = line;
                line = br.readLine();
                if (tipo.equals("trainer")) {
                    tok = new StringTokenizer(line);
                    int ore = Integer.parseInt(tok.nextToken());
                    double cost = Double.parseDouble(tok.nextToken());
                    line = br.readLine();
                    String specialita = line;

                    Trainer t = new Trainer(cod, nome, cost, ore, specialita);
                    dipendente.add(t);
                    codProg.put(cod, t);

                } else {
                    tok = new StringTokenizer(line);
                    long telefono = Long.parseLong(tok.nextToken());
                    boolean medico = Boolean.parseBoolean(tok.nextToken());
                    int appuntamenti = Integer.parseInt(tok.nextToken());
                    double cost = Double.parseDouble(tok.nextToken());

                    Nutrizionista n = new Nutrizionista(cod, nome, cost, telefono, medico, appuntamenti);
                    dipendente.add(n);
                    codProg.put(cod, n);

                }
                line = br.readLine();
            }
            br.close();
        } catch (IOException e) {
            System.err.println(e);
        } catch (Exception e) {
            System.err.println(e);
        }

        // punto 2

        try {

            BufferedReader br = new BufferedReader(new FileReader("clienti.txt"));
            String line = br.readLine();
            while (line != null) {
                int cod = Integer.parseInt(line);
                line = br.readLine();
                String name = line;
                Cliente c = new Cliente(cod, name);
                cliente.add(c);
                line = br.readLine();
                StringTokenizer tok = new StringTokenizer(line);
                while (tok.hasMoreTokens()) {
                    int codidip = Integer.parseInt(tok.nextToken());
                    double ore = Double.parseDouble(tok.nextToken());
                    line = br.readLine();
                    c.addServizio(codProg.get(codidip), ore);
                    if (line != null)
                        tok = new StringTokenizer(line);
                }
                line = br.readLine();
            }
            br.close();
        } catch (IOException e) {
            System.err.println(e);
        } catch (Exception e) {
            System.err.println(e);
        }

        // punto 3
        System.out
                .println("nome, codice, tipo, ore settimanali, specialità, telefono, medico, app.sett., costo orario");
        for (Dipendente d : dipendente) {
            System.out.println(d.toString());
        }
        System.out.println();

        // punto 4
        for (Cliente c : cliente) {
            System.out.println(c);
        }

        // punto 5
        System.out.println();
        String name = null;
        int ciaone = 0;
        for (Cliente c : cliente) {
            if (ciaone <= c.getMaximo()) {
                ciaone = c.getMaximo();
                name = c.getName();
            }

        }

        System.out.println(name + "\t" + ciaone);
    }

}

File: esercizi_python/compito_python_2/Gestione.py
import sys
import argparse


from Dipendenti import Dipendenti
from Trainer import Trainer
from Nutrizionista import Nutrizionista
from Cliente import Cliente
from Servizio import Servizio


class Gestione:
    def __init__(self, arg):
        self._arg = arg

    def main(self):
        dipendenti = []
        clienti = []
        codProg = {}

        # punto 1

        try:
            f = open("dipendenti.txt", "r")
            line = f.readline().strip()
            while(line != ''):
                tok = line.split()
                cod = int(tok[0])
                tipo = tok[1]
                line = f.readline().strip()
                nome = line
                line = f.readline().strip()
                tok = line.split()
                if tipo == "trainer":
                    ore = int(tok[0])
                    costo = float(tok[1])
                    line = f.readline().strip()
                    specialita = line

                    t = Trainer(cod, nome, costo, ore, specialita)
                    dipendenti.append(t)

                    codProg.update({cod: t})
                else:
                    tel = int(tok[0])  # ci va il long
                    medico = bool(tok[1])
                    appuntamenti = int(tok[2])
                    costo = float(tok[3])

                    n = Nutrizionista(cod, nome, costo, tel,
                                      medico, appuntamenti)
                    dipendenti.append(n)

                    codProg.update({cod: n})
                line = f.readline().strip()
            f.close()

        except IOError:
            print("IOError")
        except:
            print("Error")
            raise

        # punto 2
        try:
            f = open("clienti.txt", "r")
            line = f.readline().strip()
            while line != '':
                cod = int(line)
                line = f.readline().strip()
                nome = line
                c = Cliente(cod, nome)
                clienti.append(c)
                line = f.readline().strip()

                tok = line.split()

                while(len(tok) == 2):
                    codcliente = int(tok[0])
                    ore = float(tok[1])
                    # chiamata funzione all'interno di clienti per aggiungere
                    c.addServizio(codProg.get(codcliente), ore)

                    line = f.readline().strip()

                    tok = line.split()

                line = f.readline().strip()
            f.close()

        except IOError:
            print("IOError")
        except:
            print("Error")
            raise

        # punto 3
        print("nome, codice, tipo, ore settimanali, specialità, telefono, medico, app.sett., costo orario")
        for d in dipendenti:
            print(d.toString())

        # punto 4
        print("\n")
        for c in clienti:
            print(str(c.toString()))

        # punto 5
        a = 0
        b = 0
        ciaone = 0
        duna = ""
        nome = ""
        for d in dipendenti:
            for c in clienti:
                b = c.numServ(d.cod)
                if ciaone >= a:
                    a = a + b
                    nome = d.nome
            ciaone = a
            a = 0
            duna = nome
        print(duna + "\t" + str(ciaone))


if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument('matricola', action='store')
    #matr = parser.parse_args()
    matr = 10
    g = Gestione(matr)
    g.main()
